source/minio-features/bucket-versioning.rst

```diff
@@ -0,0 +1,87 @@
+.. _minio-bucket-versioning:
+
+=================
+Bucket Versioning
+=================
+
+.. default-domain:: minio
+
+.. contents:: On This Page
+   :local:
+   :depth: 2
+
+MinIO supports keeping multiple "versions" of an object in a single bucket.
+Write operations which would normally overwrite an existing object instead
+result in the creation of a new versioned object. MinIO versioning protects from
+unintended overwrites and deletions while providing support for "undoing" a
+write operation. Bucket versioning also supports retention and archive policies.
+
+<Diagram>
+
+MinIO generates a unique immutable ID for each object. If a ``PUT`` request
+contains an object name which duplicates an existing object, MinIO does *not*
+overwrite the "older" object. Instead, MinIO retains all object versions while
+considering the most recently written "version" of the object as "latest".
+Applications retrieve the latest object version by default, but *may* retrieve
+any other version in the history of that object. To view all versions of an
+object or objects in a bucket, use the :mc-cmd-option:`mc ls versions` command.
```

Comment:
```suggestion
object or objects in a bucket, use the :mc-cmd-option:`mc ls --versions` command.
```

Reply:
So the `:mc-cmd-option:` directive has a bit of odd behavior. It splits by whitespace and adds `--` as a suffix to the last argument (in this case, `versions`). 

This is mostly necessary due to some quirks with the way Sphinx parses whitespaces into anchorable links (it uses `-` to replace ` ` whitespace characters, so you end up with broken option links. 

I might be able to fix this later with optimizations to the underlying Python code, but that's for later.

This is all to say, this should work as designed - that is, in the rendered output, this shows as `mc ls --versions`. 

You can build locally to check, the build process is relatively straightforward (create a virtual environment, install from requirements, run make html)